### PR TITLE
COM-595 Add hiv defaulters program config

### PR DESCRIPTION
**Pull request by lauravignoli**

### CODE REVIEW

**Comment by tmvumbi2** [comment]
@lauravignoli The field "Program Outcome"  is appearing in `create` form  and is duplicated in `edit` form (this is happening on HIV and TB programs as well)

File: openmrs/apps/clinical/app.json [hunk 1]
```diff
@@ -36,34 +36,65 @@
...
-            		"class",
-            		"caste"
-        		],
-        		"additionalPatientIdentifiers": [
-           			"National ID"
+        "program": {
+            "patientInformation": {
+                "ageLimit": 100
+            },
+            "PROGRAM_MANAGEMENT_PATIENT_COMORBIDITES": {
```
Comment (tmvumbi2) [inline openmrs/apps/clinical/app.json:43]: @lauravignoli Do you know if it is a requirement to maintain the field order as in the Excel sheet?

**Comment by tmvumbi2** [comment]
@lauravignoli Does the field the field "Treatment date" correspond to  "Date of ART initiation" specified in the excel sheet? If yes, do you know if we should use the label in the sheet?

[update] similar question for "Patient clinical stage" / "Tracking state"

**Comment by tmvumbi2** [comment]
@lauravignoli The field "Tracking state" (i.e. "Patient clinical stage") needs to be `required` for the defaulter program

File: openmrs/apps/clinical/app.json [hunk 2]
```diff
@@ -36,34 +36,65 @@
...
+            "PROGRAM_MANAGEMENT_PATIENT_STAGE": {
+                "excludeFrom": ["TB_PROGRAM_KEY", "HIV_DEFAULTERS_PROGRAM_KEY"]
+            },
+            "PROGRAM_MANAGEMENT_PATIENT_DOCTOR_IN_CHARGE": {
+                "excludeFrom": ["HIV_DEFAULTERS_PROGRAM_KEY"]
+            },
+            "PROGRAM_MANAGEMENT_ART_NUMBER": {
+                "excludeFrom": ["TB_PROGRAM_KEY", "HIV_DEFAULTERS_PROGRAM_KEY"]
+            },
+            "PROGRAM_MANAGEMENT_REASON_FOR_DEFAULTING": {
```
Comment (tmvumbi2) [inline openmrs/apps/clinical/app.json:61]: @lauravignoli this field needs to be multi-select (I'm not sure if this is possible to do in the configuration)

**Comment by tmvumbi2** [comment]
@lauravignoli I noticed that the HIV program doesn't have the fields `TARV code` and `reason for consultation` anymore, just confirming if it is intentional?

File: openmrs/apps/clinical/app.json (hunk 2 above)
Comment (lauravignoli) [reply openmrs/apps/clinical/app.json:61]: I haven't found a way to do it. We might have to do yes/no for each reason. Would like to check with Lisa when she is back. I will add a comment on the ticket.

**Comment by lauravignoli** [comment]
> @lauravignoli I noticed that the HIV program doesn't have the fields `TARV code` and `reason for consultation` anymore, just confirming if it is intentional?

This is true for the TB program, and yes it was intentional. I checked with JP. Both TARV and reason for consultation fields are HIV specific.

**Comment by lauravignoli** [comment]
> @lauravignoli The field "Program Outcome" is appearing in `create` form and is duplicated in `edit` form (this is happening on HIV and TB programs as well)

I don't see Program Outcome in any of the create forms. Can you please check again?

**Comment by lauravignoli** [comment]
> @lauravignoli The field "Tracking state" (i.e. "Patient clinical stage") needs to be `required` for the defaulter program

Done here: https://github.com/jembi/cameroon-openmrs-module-bahmniapps/pull/141/commits/d96086554f555b1510d782519345f53bf616b56f

File: openmrs/apps/clinical/app.json (hunk 1 above)
Comment (lauravignoli) [reply openmrs/apps/clinical/app.json]: I don't think so

Comment (lauravignoli) [reply openmrs/apps/clinical/app.json:43]: I think let's leave it for now?

**Pull request merged**